Cython/Compiler/FusedNode.py

```diff
@@ -613,12 +633,20 @@ def make_fused_cpdef(self, orig_py_func, env, is_def):
                     int __Pyx_Is_Little_Endian()
             """)
         decl_code.indent()
-
         pyx_code.put_chunk(
             u"""
-                def __pyx_fused_cpdef(signatures, args, kwargs, defaults):
+                def __pyx_fused_cpdef(signatures, args, kwargs, defaults):#, *, _fused_sigindex=(None,)):
                     # FIXME: use a typed signature - currently fails badly because
                     #        default arguments inherit the types we specify here!
+                    # FIXME: Keep the signature index in a local default keyword argument
+                    #        to make it persist through calls without polluting global
+                    #        namespace. Currently results in segfault when resulting
+                    #        function is later bound as a method.
+                    #        https://github.com/cython/cython/issues/3370
+                    # FIXME: Avoid using a mutable hidden default keyword argument
+                    #        to make a local object persist through different function calls.
+                    #        (Reconstructing it every call (even from a literal)
+                    #        would probably incur a cost scaled to its `O(k**n)` size.)
```

Comment:
Is using the global constant for the specialization indices an acceptable permanent solution for making them persistent?

The other options are probably:

1) Move them back into the function as a hidden keyword argument, maybe with a default value at compile time if possible. This feels quite hack-y to me, though. Aside from #3370, arguments are meant for passing values *into* the function, and using them to keep what is basically a local constant via a mutable object at definition time doesn't seem like a good reason to change the function's signature.

2) Make `cpdef` function dispatchers fully-fledged objects (or object-like structures) and keep the signature as an attribute. This seems like it'd add a fair bit of overhead, in terms of both development and run time.

With the properly done name mangling, and since there does seem to be a precedent for global Cython-only values in this setting (like `__pyx_unpickle_compiled_fused_type_cls` and `__pyx_capi__`), keeping the global seems like it might be the cleanest option to me.

Reply:
I don't think there's a problem with this (but I'm not the person who gets the final say).

The real worry is if the variable accidentally ends up being leaked back to the user. It certainly doesn't appear in `globals()` or `dir(module)`. You can get it by adding to the Cython module:

```
def hmmm():
    return __pyx_fusedcpdef_sigindices
```
but that seems a fairly small leak because you do need to know what you're looking for.